File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.java
// Last updated: 7/2/2025, 11:53:45 PM
class Solution {
    public int findMin(int[] nums) {
        
        int min = nums[0];
      int l = 0;
        int h = nums.length-1;
        while(l<h)
        {
            int mid = (l+h)/2;
            //min = nums[mid];
            if(nums[mid]<nums[h]) h = mid;
            else l = mid+1;
        }
        return nums[l];
    }
}

File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.py
# Last updated: 7/2/2025, 11:53:41 PM
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)        
        l, r = 0, n-1
        res = nums[0]
        while(l<r):
            mid = (l + r) // 2

            if(nums[mid] > nums[r]):
                l = mid+1
            else:
                r = mid
            
        return nums[l]
        
        